app/src/main/java/com/example/weather/LocationClient.java

```diff
@@ -0,0 +1,103 @@
+package com.example.weather;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.location.Location;
+import android.location.LocationListener;
+import android.os.Looper;
+import android.os.Handler;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.core.app.ActivityCompat;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.location.LocationResult;
+import com.google.android.gms.location.LocationServices;
+import com.google.android.gms.location.LocationCallback;
+
+public class LocationClient {
+
+    private static final String TAG = LocationClient.class.getSimpleName();
+
+    private FusedLocationProviderClient fusedLocationClient;
+    private LocationRequest locationRequest;
+    private LocationCallback locationCallback;
+
+    private double latitude;
+    private double longitude;
```

Comment:
Create model class Location and have it there.
```suggestion
    private Location location;
```